frame/session/src/lib.rs

```diff
@@ -30,92 +30,75 @@
-//! key for the length of the session.
-//! - **Session length change process:** At the start of the next session we allocate a session index and record the
-//! timestamp when the session started. If a `NextSessionLength` was recorded in the previous session, we record
-//! it as the new session length. Additionally, if the new session length differs from the length of the
-//! next session then we record a `LastLengthChange`.
+//! - **Session length:** This pallet does not assume anything about the length of each session.
+//! Rather, it relies on an implementation of `ShouldEndSession` to dictate a new session's start.
+//! This pallet provides the `PeriodicSessions` struct for simple periodic sessions.
 //! - **Session rotation configuration:** Configure as either a 'normal' (rewardable session where rewards are
 //! applied) or 'exceptional' (slashable) session rotation.
-//! - **Session rotation process:** The session is changed at the end of the final block of the current session
-//! using the `on_finalize` method. It may be called by either an origin or internally from another runtime
-//! module at the end of each block.
+//! - **Session rotation process:** At the beginning of each block, the `on_initialize` function
+//! queries the provided implementation of `ShouldEndSession`. If the session is to end the newly
+//! activated validator IDs and session keys are taken from storage and passed to the
+//! `SessionHandler`. The validator set supplied by `OnSessionEnding` and the corresponding session
+//! keys, which may have been registered via `set_keys` during the previous session, are written ////! to storage where they will wait one session before being passed to the `SessionHandler`
+//! themselves.
 //!
 //! ### Goals
 //!
-//! The Session module in Substrate is designed to make the following possible:
+//! The Session pallet is designed to make the following possible:
 //!
-//! - Set session keys of the validator set for the next session.
-//! - Set the length of a session.
+//! - Set session keys of the validator set for upcoming sessions.
+//! - Control the length of sessions.
 //! - Configure and switch between either normal or exceptional session rotations.
 //!
 //! ## Interface
 //!
 //! ### Dispatchable Functions
 //!
-//! - `set_key` - Set a validator's session key for the next session.
-//! - `set_length` - Set a new session length to be applied upon the next session change.
-//! - `force_new_session` - Force a new session that should be considered either a normal (rewardable)
-//! or exceptional rotation.
-//! - `on_finalize` - Called when a block is finalized. Will rotate session if it is the last
-//! block of the current session.
+//! - `set_keys` - Set a validator's session keys for upcoming sessions.
+//! - `on_initialize` - Called when a block is initialized. Will rotate session if it is the first
+//! block of the new session.
```

Comment:
on_initialize is not really a dispatchable function is it ? but if the docs are consistent I'm ok.

Reply:
Yeah, I thought that too, and left it because `on_finalize` was in there before. I'll remove it though. I think it makes more sense being removed.